discovery-client/src/main/java/io/micronaut/discovery/spring/config/SpringCloudClientConfiguration.java

```diff
@@ -113,6 +115,20 @@ public Optional<String> getName() {
         return name != null ? Optional.of(name) : Optional.empty();
     }
 
+    /**
+     * @return The spring cloud config username.
+     */
+    public Optional<String> getUsername() {
+        return username != null ? Optional.of(username) : Optional.empty();
+    }
+
+    /**
+     * @return The spring cloud config password.
+     */
+    public Optional<String> getPassword() {
+        return password != null ? Optional.of(password) : Optional.empty();
```

Comment:
please change this to `Optional.ofNullable(password)` and remove the ternary operator